kolibri/plugins/learn/assets/src/views/LearnImmersiveLayout.vue

```diff
@@ -219,6 +267,105 @@
       });
     },
     methods: {
+      getSidebarInfo() {
+        if (this.lessonId) {
+          this.fetchLessonSiblings();
+        } else {
+          if (this.content.id) {
+            this.fetchSiblings();
+          }
+        }
+      },
+      /** Returns a promise which will get the progress for a set of nodes and map it back onto
+       *  the nodes, which are then set to this.viewResourcesContents.
+       *  @modifies this.viewResourcesContents - Sets it to the value of nodes, mapped with
+       *  progress, if found
+       *  @returns {Promise} - Just returns a meaningless Promise in case you want to chain
+       *  synchronously
+       */
+      progressPromise(nodes) {
+        // Avoid unnecessary fetches, set values, return empty Promise
+        if (!this.$store.getters.isUserLoggedIn) {
+          nodes.map(node => (node.progress = 0));
+          this.viewResourcesContents = nodes;
+          return Promise.resolve();
+        }
+        const getParams = {
+          content_ids: nodes.map(node => node.content_id),
+        };
+        return ContentNodeProgressResource.fetchCollection({ getParams }).then(progresses => {
+          this.viewResourcesContents = nodes
+            .map(node => {
+              const matchingProgress = progresses.find(p => p.content_id === node.content_id) || {
+                progress: 0,
+              };
+              node.progress = matchingProgress.progress;
+              return node;
+            })
+            .filter(node => node.content_id !== this.content.content_id);
+        });
+      },
+      /**
+       * Prepares a list of content nodes which are also in the same lesson as this.lessonId
+       * @modifies this.viewResourcesContents - Sets it to the progress-mapped nodes
+       */
+      fetchLessonSiblings() {
+        LessonResource.fetchModel({ id: this.lessonId }).then(lesson => {
+          ContentNodeResource.fetchCollection({
+            getParams: {
+              ids: lesson.resources.map(resource => resource.contentnode_id),
+            },
+          }).then(contentNodes => {
+            return this.progressPromise(contentNodes);
+          });
+        });
+      },
+      /**
+       * Prepares a list of content nodes which are children of this.content.parent without
+       * this.content. The prepared nodes have their progress values for the current user
+       * (if there is one) mapped onto each node.
+       *
+       * Largely borrowed from modules/topicsTree/handlers.js
+       *
+       * @modifies this.viewResourcesContents - Sets it to the progress-mapped nodes
+       * @modifies this.nextFolder - Sets the value with this.content's parents next sibling folder
+       * if found
+       */
+      fetchSiblings() {
+        // Guard against this in lesson context
+        if (this.lessonId) {
+          return;
+        }
+
```

Comment:
No need to wait for this fetch to resolve, can use the `fetchContentNodeTreeProgress` with the same parameters: https://github.com/learningequality/kolibri/blob/release-v0.15.x/kolibri/plugins/learn/assets/src/composables/useContentNodeProgress.js#L50

Another thing to note is that the fetchTree response is paginated by default, so if it's a big topic, it won't contain everything.